src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -509,11 +492,11 @@ private static Socket Connect(string serverName, int port, TimeoutTimer timeout,
                         return socket;
                     }
                 }
-                catch (AggregateException aggregateException) when (aggregateException.InnerException is SocketException socketException)
+                catch (SocketException e)
```

Comment:
Now that you have removed the capability of timing out when we fail to get an open socket at the end of the timeout duration (which was enforced by a Task timeout), would it make sense to check if the socket is not connected and we have exceeded the timeout. Hence throw an exception?

Reply:
I had a hard time groking the logic here. But it does time out with an error. When socket.Blocking is set to true (line 413), socket.Connect (line 429) will throw a SocketException immediately and socketException.SocketErrorCode  will be SocketError.WouldBlock. We loop inside the catch where socket.Select (line 469) does accept a timeout. socket.Select will proceed either at a successful connect (something selected) or timeout. If timed out, the selected list will be empty and the loop repeats. If the timeout is expired, we return a null Socket (line 455), which is handled (report SNI error) in the calling methods.

Reply:
@saurabh500 when we have  `!isInfiniteTimeout` no matter what we throw an exception of type SocketException with error code wouldBlock then inside catch block Socket.Select will check for the timeout expiration. This was suggested by @roji [here](https://github.com/dotnet/SqlClient/issues/826#issuecomment-736224118) since Socket.Connect (sync) does not support timeout.

Reply:
Ah got it. Then why did we need the original change to timeout with Tasks? Is it to have more high fidelity timeout? 

> When socket.Blocking is set to true (line 413), socket.Connect (line 429) will throw a SocketException immediately

Did you mean this to be the other way around? socket.Blocking = false should lead to an immediate exception. Or that's what I understand. This means that "hey socket do not block any calls, but keep trying and I will come back and check later using Select", and the socket responds by throwing an exception immediately with the error code specifying the reason, so that the caller can omit the exception, and move on to the select.

Reply:
@JRahnama  I understand the reasoning behind teh SocketException with WouldBlock and that makes sense to me. My question was different

Reply:
> Ah got it. Then why did we need the original change to timeout with Tasks? Is it to have more high fidelity timeout?

My assumption is the previous changes were made because of a misunderstanding of the logic, thinking it wasn't timing out. The original PR included many other changes improving the timeout logic in SNI.

> 
> > When socket.Blocking is set to true (line 413), socket.Connect (line 429) will throw a SocketException immediately
> 
> Did you mean this to be the other way around? socket.Blocking = false should lead to an immediate exception. Or that's what I understand. This means that "hey socket do not block any calls, but keep trying and I will come back and check later using Select", and the socket responds by throwing an exception immediately with the error code specifying the reason, so that the caller can omit the exception, and move on to the select.

Oops, yes. Fixed.